interfaces/builtin/network_manager_observe.go

```diff
@@ -0,0 +1,146 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"strings"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/release"
+	"github.com/snapcore/snapd/snap"
+)
+
+const networkManagerObserveBaseDeclarationSlots = `
+  network-manager-observe:
+    allow-installation:
+      slot-snap-type:
+        - app
+        - core
+    deny-auto-connection: true
+    deny-connection:
+      on-classic: false
+`
+
+const networkManagerObserveSummary = `allows observing NetworkManager settings`
+
+const networkManagerObserveConnectedSlotAppArmor = `
```

Comment:
NM sends a couple of signals that could be useful too: org.freedesktop.NetworkManger.PropertiesChanged and org.freedesktop.NetworkManager.StateChanged on /org/freedesktop/NetworkManger. Note that org.freedesktop.NetworkManger.PropertiesChanged  isn't exactly the same as freedeskto.DBus.Properties.PropertiesChanged, but while at it we may want to allow both.

There is also a couple of potentially useful signals in the Settings interface (/org/freedesktop/NetworkManager/Settings path)

Reply:
Looked at `gio`, the `GNetworkMonitor` backend for NetworkManager uses `org.freedesktop.NetworkManger.PropertiesChanged` signal to trigger the refresh of a bunch of variables.

Reply:
Ok, I added the corresponding signal rules for allowed method rules. I'm told this is all that is required by QtNetworkBearer; we can iterate on the interface as needed (but the signals you mentioned should be covered).

Reply:
> NM sends a couple of signals that could be useful too: org.freedesktop.NetworkManger.PropertiesChanged and org.freedesktop.NetworkManager.StateChanged on /org/freedesktop/NetworkManger. Note that org.freedesktop.NetworkManger.PropertiesChanged isn't exactly the same as freedeskto.DBus.Properties.PropertiesChanged, but while at it we may want to allow both.
> 
> There is also a couple of potentially useful signals in the Settings interface (/org/freedesktop/NetworkManager/Settings path)

@bboozzoo - this is done.